streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImplTest.java

```diff
@@ -53,19 +57,27 @@ public void testRemoveSensor() {
         String entity = "entity";
         String operation = "put";
         Map<String, String> tags = new HashMap<>();
-        StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(new Metrics(), groupName, tags);
+        final Metrics metrics = new Metrics();
+        final Map<MetricName, KafkaMetric> initialMetrics = Collections.unmodifiableMap(new LinkedHashMap<>(metrics.metrics()));
```

Comment:
Do we need to wrap with the LinkedHashMap? Could we just do `Collections.unmodifiableMap(metrics.metrics());`

Reply:
I can look into it. I thought that `unmodifiableMap` only wraps the collection, not copies it, and we need to make a copy here.

Reply:
Yep, it's only a view, and we need a copy: "Returns an unmodifiable view of the specified map."